lib/ohai/plugins/solaris2/network.rb

```diff
@@ -86,7 +86,7 @@ def arpname_to_ifname(iface, arpname)
 
   def full_interface_name(iface, part_name, index)
     iface.each do |name, attrs|
-      next unless attrs && attrs.respond_to?(:[])
+      next unless attrs&.respond_to?(:[])
```

Comment:
Even `nil` responds to `respond_to?`.

Reply:
Oh yep that should be a cop right there

Reply:
https://github.com/rubocop-hq/rubocop/issues/8668